modules/Blockchain/Ethereum/index.js

```diff
@@ -1547,6 +1593,13 @@ class Ethereum {
             return gasPrice;
         }
     }
+
+    numberOfEventsEmmitted(receipt) {
+        if (!receipt || !receipt.logs || !Array.isArray(receipt.logs)) {
+            return undefined;
```

Comment:
Are we expecting number or undefined on the caller side?

Reply:
Yeah, whichever it returns we've got it handled, but I'll add some documentation here so it's clearer